.github/workflows/docker_images_template.yaml

```diff
@@ -23,37 +25,27 @@ jobs:
           ref: ${{ github.ref }}
           head_ref: ${{ github.head_ref }}
 
-      - name: Build image
-        env:
-          PYPI_TAG: ${{steps.get-ref.outputs.tag}}
-        run: |
-          echo "Building service: ${{inputs.wmcore_component}}, with tag: ${PYPI_TAG}"
-          svn checkout https://github.com/dmwm/CMSKubernetes/trunk/docker/pypi/${{inputs.wmcore_component}}
-          cd ${{inputs.wmcore_component}}
-          cat Dockerfile
-          echo "Sleeping 5min to ensure that PyPi packages are available..."
-          sleep 300
-          docker build --build-arg TAG=${PYPI_TAG} --tag registry.cern.ch/cmsweb/${{inputs.wmcore_component}}:${PYPI_TAG} .
-
-      - name: Images
-        run: |
-          docker images
-
       - name: Login to registry.cern.ch
         uses: docker/login-action@v2
         with:
           registry: registry.cern.ch
           username: ${{ secrets.cern_user }}
           password: ${{ secrets.cern_token }}
 
-      - name: Publish image to registry.cern.ch
-        uses: docker/build-push-action@v1
-        with:
-          path: ${{inputs.wmcore_component}}
-          build_args: |
-            TAG=${{steps.get-ref.outputs.tag}}
-          registry: registry.cern.ch
-          username: ${{ secrets.cern_user }}
-          password: ${{ secrets.cern_token }}
-          repository: cmsweb/${{inputs.wmcore_component}}
-          tag_with_ref: true
+      - name: Build and publish docker image
+        env:
+          PYPI_TAG: ${{steps.get-ref.outputs.tag}}
+          CERN_REGISTRY: registry.cern.ch
+        run: |
+          echo "Building service: ${SERVICE_NAME}, with tag: ${PYPI_TAG}"
+          svn checkout https://github.com/dmwm/CMSKubernetes/trunk/docker/pypi/${SERVICE_NAME}
```

Comment:
Why do we need `svn` to checkout github repo? Why not to use `git` for that?

Reply:
For this question, please refer to: https://github.com/dmwm/WMCore/pull/11639 for full context.

Reply:
@todor-ivanov has done the research on the `svn` util. My understanding is that with `svn`, we can fetch a sub-directory of the repository, while `git` does not provide this capability and we would have to clone the whole repository for each of the 11(?) services that we have setup in the actions workflow.

Reply:
ok, thanks for providing the reason. Said that, this line is not doing what the comment is saying, i.e. it will fetch latest subdir and not a particular tag. I do not know `svn` enough to answer if it can fetch proper tag of sub-dir, but `curl` cat fetch tarball of particular tag, and `git` indeed will fetch entire repo for that tag. So, we should fix either `svn` or switch back to `ciurl` to fetch exact tag of the repo.

Reply:
Valentin, the CMSKubernetes code that we check out isn't really tagged (we always fetch what is in HEAD).
The PYPI_TAG is indeed used, but then in the scope of WMCore (used for the tag in pypi and the docker build argument).

Having said that, I would say it works as expected and there is nothing to be changed here.

Reply:
Hi @vkuznet @amaltaro 
> with svn, we can fetch a sub-directory of the repository, while git does not provide this capability and we would have to clone the whole repository for each of the 11(?) services that we have setup

Yes this is the exact reason. 

> So, we should fix either svn or switch back to curl to fetch exact tag of the repo

SVN is fully capable of fetching  particular tag.